### PR TITLE
WT-6917 Create WiredTiger contributing guide

### PR DESCRIPTION
**Pull request by agorrod**
This PR adds contributor information to WiredTiger repository, it mostly links to a wiki page with more content.

### CODE REVIEW

File: CONTRIBUTING [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+Contributing to the WiredTiger project
+======================================
+
+Pull requests are always welcome, and the WiredTiger dev team appreciates any help the community can
+give to help make WiredTiger better.
+
+For more information about how to contribute, please read `the WiredTiger Wiki on GitHub`_.
+
+.. _the WiredTiger Wiki on GitHub: https://github.com/wiredtiger/wiredtiger/wiki/Contributing-to-WiredTiger
```
Comment (korteland) [inline CONTRIBUTING:10]: Is this `.. _` thing the syntax for a footnote? I tried it with a couple of online renderers and it just rendered as-is.

**Review by korteland — APPROVED** [review]
LGTM, one question about markdown syntax.

**Review by mickgraham-mongodb — APPROVED** [review]

**Review by lukech — APPROVED** [review]
LGTM (modulo the question raised by Will).

File: CONTRIBUTING (hunk 1 above)
Comment (agorrod) [reply CONTRIBUTING:10]: Ah - I'm not sure, I copied that from the MongoDB contributor guide verbatim, so I figured it was some odd syntax. Let me see if I can get more confident that it's correct.

**Comment by agorrod** [comment]
I'm going to go ahead and merge - I can't figure out how to get more confidence in the link working before doing that.

It is compliant with the RST format documentation here: https://en.wikipedia.org/wiki/ReStructuredText

**Comment by korteland** [comment]
Ah I didn't realise it was RST. Thanks for explaining that.

**Pull request merged**